enlightenment.efl: build failure with gcc15

### Nixpkgs version

- Unstable (26.05)

### Describe the bug

GCC15 update broke efl build.

### Steps to reproduce

``` console
$ nix-build --option sandbox true -K -A enlightenment.efl
```

### Expected behaviour

Should build successfully.

### Screenshots

_No response_

### Relevant log output

```console
console
[335/2527] Compiling C object src/lib/emile/libemile.so.1.28.1.p/emile_image.c.o
FAILED: [code=1] src/lib/emile/libemile.so.1.28.1.p/emile_image.c.o 
gcc -Isrc/lib/emile/libemile.so.1.28.1.p -Isrc/lib/emile -I../src/lib/emile -I. -I.. -Isrc/lib/eina -I../src/lib/eina -Isrc/lib/efl -I../src/lib/efl -Isrc/lib/eo -I../src/lib/eo -Isrc/static_libs/rg_etc -I../src/static_libs/rg_etc -Isrc/lib/efl/interfaces -I/nix/store/5ljn72xffsivv2ya5ncrzkyf5nx94227-libjpeg-turbo-3.1.3-dev/include -I/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include -I/nix/store/srnvjrr38j7xmpcmjpndd6mw31lsfjrl-zlib-1.3.1-dev/include -I/nix/store/sl10j863s32x7dv5vw6wfx1nh22f3xh2-lz4-1.10.0-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -fvisibility=hidden -Wpointer-arith -Wunused-parameter -Wsign-compare -Wno-missing-field-initializers -DHAVE_CONFIG_H=1 -D_GNU_SOURCE=1 -DEFL_BETA_API_SUPPORT=1 -DELM_INTERNAL_API_ARGESFSDFEFC=1 -fPIC -pthread '-DPACKAGE_DATA_DIR="/nix/store/g1liv2kld9f44b647pmq8iax4r6a0f97-efl-1.28.1/share/emile"' -DNEED_RUN_IN_TREE=1 -DEFL_BUILD=1 -MD -MQ src/lib/emile/libemile.so.1.28.1.p/emile_image.c.o -MF src/lib/emile/libemile.so.1.28.1.p/emile_image.c.o.d -o src/lib/emile/libemile.so.1.28.1.p/emile_image.c.o -c ../src/lib/emile/emile_image.c
In file included from ../src/lib/emile/emile_image.c:16:
../src/static_libs/rg_etc/rg_etc1.h:6:23: error: 'bool' cannot be defined via 'typedef'
    6 | typedef unsigned char bool;
      |                       ^~~~
../src/static_libs/rg_etc/rg_etc1.h:6:23: note: 'bool' is a keyword with '-std=c23' onwards
../src/static_libs/rg_etc/rg_etc1.h:6:1: warning: useless type name in empty declaration
    6 | typedef unsigned char bool;
      | ^~~~~~~
[336/2527] Compiling C object src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_etc2_encoder.c.o
FAILED: [code=1] src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_etc2_encoder.c.o 
gcc -Isrc/lib/emile/libemile.so.1.28.1.p -Isrc/lib/emile -I../src/lib/emile -I. -I.. -Isrc/lib/eina -I../src/lib/eina -Isrc/lib/efl -I../src/lib/efl -Isrc/lib/eo -I../src/lib/eo -Isrc/static_libs/rg_etc -I../src/static_libs/rg_etc -Isrc/lib/efl/interfaces -I/nix/store/5ljn72xffsivv2ya5ncrzkyf5nx94227-libjpeg-turbo-3.1.3-dev/include -I/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include -I/nix/store/srnvjrr38j7xmpcmjpndd6mw31lsfjrl-zlib-1.3.1-dev/include -I/nix/store/sl10j863s32x7dv5vw6wfx1nh22f3xh2-lz4-1.10.0-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -fvisibility=hidden -Wpointer-arith -Wunused-parameter -Wsign-compare -Wno-missing-field-initializers -DHAVE_CONFIG_H=1 -D_GNU_SOURCE=1 -DEFL_BETA_API_SUPPORT=1 -DELM_INTERNAL_API_ARGESFSDFEFC=1 -fPIC -pthread '-DPACKAGE_DATA_DIR="/nix/store/g1liv2kld9f44b647pmq8iax4r6a0f97-efl-1.28.1/share/emile"' -DNEED_RUN_IN_TREE=1 -DEFL_BUILD=1 -MD -MQ src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_etc2_encoder.c.o -MF src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_etc2_encoder.c.o.d -o src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_etc2_encoder.c.o -c ../src/static_libs/rg_etc/etc2_encoder.c
In file included from ../src/static_libs/rg_etc/etc2_encoder.c:34:
../src/static_libs/rg_etc/rg_etc1.h:6:23: error: 'bool' cannot be defined via 'typedef'
    6 | typedef unsigned char bool;
      |                       ^~~~
../src/static_libs/rg_etc/rg_etc1.h:6:23: note: 'bool' is a keyword with '-std=c23' onwards
../src/static_libs/rg_etc/rg_etc1.h:6:1: warning: useless type name in empty declaration
    6 | typedef unsigned char bool;
      | ^~~~~~~
[337/2527] Compiling C object src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc1.c.o
FAILED: [code=1] src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc1.c.o 
gcc -Isrc/lib/emile/libemile.so.1.28.1.p -Isrc/lib/emile -I../src/lib/emile -I. -I.. -Isrc/lib/eina -I../src/lib/eina -Isrc/lib/efl -I../src/lib/efl -Isrc/lib/eo -I../src/lib/eo -Isrc/static_libs/rg_etc -I../src/static_libs/rg_etc -Isrc/lib/efl/interfaces -I/nix/store/5ljn72xffsivv2ya5ncrzkyf5nx94227-libjpeg-turbo-3.1.3-dev/include -I/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include -I/nix/store/srnvjrr38j7xmpcmjpndd6mw31lsfjrl-zlib-1.3.1-dev/include -I/nix/store/sl10j863s32x7dv5vw6wfx1nh22f3xh2-lz4-1.10.0-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -fvisibility=hidden -Wpointer-arith -Wunused-parameter -Wsign-compare -Wno-missing-field-initializers -DHAVE_CONFIG_H=1 -D_GNU_SOURCE=1 -DEFL_BETA_API_SUPPORT=1 -DELM_INTERNAL_API_ARGESFSDFEFC=1 -fPIC -pthread '-DPACKAGE_DATA_DIR="/nix/store/g1liv2kld9f44b647pmq8iax4r6a0f97-efl-1.28.1/share/emile"' -DNEED_RUN_IN_TREE=1 -DEFL_BUILD=1 -MD -MQ src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc1.c.o -MF src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc1.c.o.d -o src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc1.c.o -c ../src/static_libs/rg_etc/rg_etc1.c
In file included from ../src/static_libs/rg_etc/rg_etc1.c:23:
../src/static_libs/rg_etc/rg_etc1.h:6:23: error: 'bool' cannot be defined via 'typedef'
    6 | typedef unsigned char bool;
      |                       ^~~~
../src/static_libs/rg_etc/rg_etc1.h:6:23: note: 'bool' is a keyword with '-std=c23' onwards
../src/static_libs/rg_etc/rg_etc1.h:6:1: warning: useless type name in empty declaration
    6 | typedef unsigned char bool;
      | ^~~~~~~
[338/2527] Compiling C object src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc2.c.o
FAILED: [code=1] src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc2.c.o 
gcc -Isrc/lib/emile/libemile.so.1.28.1.p -Isrc/lib/emile -I../src/lib/emile -I. -I.. -Isrc/lib/eina -I../src/lib/eina -Isrc/lib/efl -I../src/lib/efl -Isrc/lib/eo -I../src/lib/eo -Isrc/static_libs/rg_etc -I../src/static_libs/rg_etc -Isrc/lib/efl/interfaces -I/nix/store/5ljn72xffsivv2ya5ncrzkyf5nx94227-libjpeg-turbo-3.1.3-dev/include -I/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include -I/nix/store/srnvjrr38j7xmpcmjpndd6mw31lsfjrl-zlib-1.3.1-dev/include -I/nix/store/sl10j863s32x7dv5vw6wfx1nh22f3xh2-lz4-1.10.0-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -fvisibility=hidden -Wpointer-arith -Wunused-parameter -Wsign-compare -Wno-missing-field-initializers -DHAVE_CONFIG_H=1 -D_GNU_SOURCE=1 -DEFL_BETA_API_SUPPORT=1 -DELM_INTERNAL_API_ARGESFSDFEFC=1 -fPIC -pthread '-DPACKAGE_DATA_DIR="/nix/store/g1liv2kld9f44b647pmq8iax4r6a0f97-efl-1.28.1/share/emile"' -DNEED_RUN_IN_TREE=1 -DEFL_BUILD=1 -MD -MQ src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc2.c.o -MF src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc2.c.o.d -o src/lib/emile/libemile.so.1.28.1.p/.._.._static_libs_rg_etc_rg_etc2.c.o -c ../src/static_libs/rg_etc/rg_etc2.c
In file included from ../src/static_libs/rg_etc/rg_etc2.c:41:
../src/static_libs/rg_etc/rg_etc1.h:6:23: error: 'bool' cannot be defined via 'typedef'
    6 | typedef unsigned char bool;
      |                       ^~~~
../src/static_libs/rg_etc/rg_etc1.h:6:23: note: 'bool' is a keyword with '-std=c23' onwards
../src/static_libs/rg_etc/rg_etc1.h:6:1: warning: useless type name in empty declaration
    6 | typedef unsigned char bool;
      | ^~~~~~~
[339/2527] Compiling C object src/lib/emile/libemile.so.1.28.1.p/emile_base64.c.o
[340/2527] Compiling C object src/lib/eo/libeo.so.1.28.1.p/eo_base_class.c.o
[341/2527] Compiling C object src/lib/emile/libemile.so.1.28.1.p/emile_main.c.o
[342/2527] Compiling C object src/lib/emile/libemile.so.1.28.1.p/emile_cipher_openssl.c.o
../src/lib/emile/emile_cipher_openssl.c: In function 'emile_cipher_server_listen':
../src/lib/emile/emile_cipher_openssl.c:346:4: warning: 'DH_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  346 |    dh_params = DH_new();
      |    ^~~~~~~~~
In file included from /nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include/openssl/dsa.h:31,
                 from /nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include/openssl/x509.h:37,
                 from /nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include/openssl/ssl.h:32,
                 from ../src/lib/emile/emile_cipher_openssl.c:9:
/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include/openssl/dh.h:210:27: note: declared here
  210 | OSSL_DEPRECATEDIN_3_0 DH *DH_new(void);
      |                           ^~~~~~
../src/lib/emile/emile_cipher_openssl.c:348:4: warning: 'DH_generate_parameters_ex' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  348 |    if (!DH_generate_parameters_ex(dh_params, 1024, DH_GENERATOR_5, NULL))
      |    ^~
/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include/openssl/dh.h:223:27: note: declared here
  223 | OSSL_DEPRECATEDIN_3_0 int DH_generate_parameters_ex(DH *dh, int prime_len,
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~
../src/lib/emile/emile_cipher_openssl.c:350:4: warning: 'DH_check' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  350 |    if (!DH_check(dh_params, &dh))
      |    ^~
/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include/openssl/dh.h:231:27: note: declared here
  231 | OSSL_DEPRECATEDIN_3_0 int DH_check(const DH *dh, int *codes);
      |                           ^~~~~~~~
../src/lib/emile/emile_cipher_openssl.c:354:4: warning: 'DH_generate_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  354 |    if (!DH_generate_key(dh_params))
      |    ^~
/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include/openssl/dh.h:234:27: note: declared here
  234 | OSSL_DEPRECATEDIN_3_0 int DH_generate_key(DH *dh);
      |                           ^~~~~~~~~~~~~~~
../src/lib/emile/emile_cipher_openssl.c:359:4: warning: 'DH_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  359 |    DH_free(dh_params);
      |    ^~~~~~~
/nix/store/2ivy0r8ab3bnps5957vfrxcjfcgad661-openssl-3.6.0-dev/include/openssl/dh.h:211:28: note: declared here
  211 | OSSL_DEPRECATEDIN_3_0 void DH_free(DH *dh);
      |                            ^~~~~~~
[343/2527] Compiling C object src/lib/ecore/libecore.so.1.28.1.p/efl_app.c.o
[344/2527] Compiling C object src/bin/edje/epp/epp.p/cpplib.c.o
[345/2527] Compiling C object src/lib/ecore/libecore.so.1.28.1.p/ecore.c.o
[346/2527] Compiling C object src/lib/efl/libefl.so.1.28.1.p/interfaces_efl_interfaces_main.c.o
ninja: build stopped: subcommand failed.
note: keeping build directory '/nix/var/nix/builds/nix-6462-1273414033/build'
```

### Additional context

Related: https://github.com/NixOS/nixpkgs/issues/475479

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.4, NixOS, 26.05 (Yarara), 26.05pre925861.ffbc9f8cbaac`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/a0rqwxsihszvhj4yjs05j1pg7xrp65l2-nixos/nixos`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/desktops/enlightenment/efl/default.nix
+++ b/pkgs/desktops/enlightenment/efl/default.nix
@@ -161,6 +161,9 @@ stdenv.mkDerivation rec {
 
   dontDropIconThemeCache = true;
 
+  # Fix build with gcc15 (-std=gnu23)
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isGNU "-std=gnu17";
+
   mesonFlags = [
     "--buildtype=release"
     "-D build-tests=false" # disable build tests, which are not working

